File: codeforces/p69.py
from math import sqrt

def isprime(n):
    if n == 1:
        return False
    if n == 2 or n == 3:
        return True
    if n%2 == 0 or n%3 == 0:
        return False
    for i in range(5, int(sqrt(n))+1, 6):
        if n%i == 0 or n%(i+2) == 0:
            return False
    return True

def tprime(n):
    st = sqrt(n)
    if st != int(st):
        return False
    return isprime(st)

n = int(input())
arr = [int(x) for x in input().split()]
for x in arr:
    if tprime(x):
        print("YES")
    else:
        print("NO")

File: codeforces/p69.cpp
#include <bits/stdc++.h>
using namespace std;

typedef long long ll;

bool isprime(ll n){
    if(n == 1){
        return false;
    }
    if(n == 2 || n == 3){
        return true;
    }
    if(n%2 == 0 || n%3 == 0){
        return false;
    }
    for(ll i = 5; i <= sqrt(n); i+=6){
        if(n%i == 0 || n%(i+2) == 0){
            return false;
        }
    }
    return true;
}

bool tprime(ll n){
    double sq = sqrt(n);
    if((ll) sq != sq){
        return false;
    }
    return isprime(sq);
}

int main(){
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    cout.tie(NULL);

    ll n;
    cin >> n;
    for(ll i = 0; i < n; i++){
        ll num;
        cin >> num;
        if(tprime(num)){
            cout << "YES";
        }
        else cout << "NO";
        cout << endl;
    }

}